File: app/src/test/java/com/mgaetan89/showsrage/databinding/Setters_SetTextColorResTest.java
package com.mgaetan89.showsrage.databinding;

import android.content.Context;
import android.content.res.Resources;
import android.widget.TextView;

import com.mgaetan89.showsrage.R;

import org.junit.Test;

import static org.mockito.Matchers.anyInt;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class Setters_SetTextColorResTest {
	@Test
	public void setTextColorRes() {
		Context context = mock(Context.class);
		when(context.getResources()).thenReturn(mock(Resources.class));

		TextView textView = mock(TextView.class);
		when(textView.getContext()).thenReturn(context);

		Setters.setTextColorRes(textView, R.color.primary);

		verify(textView).setTextColor(anyInt());
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/databinding/Setters_SetTextColorResTest.kt
package com.mgaetan89.showsrage.databinding

import android.content.Context
import android.content.res.Resources
import android.widget.TextView
import com.mgaetan89.showsrage.R
import org.junit.Test
import org.mockito.ArgumentMatchers.anyInt
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify

class Setters_SetTextColorResTest {
    @Test
    fun setTextColorRes() {
        val context = mock(Context::class.java)
        `when`(context.resources).thenReturn(mock(Resources::class.java))

        val textView = mock(TextView::class.java)
        `when`(textView.context).thenReturn(context)

        setTextColorRes(textView, R.color.primary)

        verify(textView).setTextColor(anyInt())
    }
}
